src/UserAction/Generic.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+// vim:ts=4:sw=4:et:fdm=marker:fdl=0
+
+namespace atk4\data\UserAction;
+
+use atk4\core\DIContainerTrait;
+use atk4\core\Exception;
+use atk4\core\InitializerTrait;
+use atk4\core\TrackableTrait;
+
+/**
+ * Implements generic user action. Assigned to a model it can be invoked by a user. Action describes meta information about
+ * the action that will help UI/API or add-ons to display action trigger (button) correctly, ensure that arguments
+ * are provided.
+ *
+ * Action must NOT rely on any specific UI implementation.
+ */
+class Generic
+{
+    use DIContainerTrait;
+    use TrackableTrait;
+    use InitializerTrait {
+        init as init_;
+    }
+
+    /** Defining scope of the action */
+    const NO_RECORDS = 'none'; // e.g. add
```

Comment:
why in place of using const and add methods that are not used, use some interface to classify the usage ?
I mean in place of a "sort of abstract" class Generic, add something like this :
- Action/iPrevieweable
- Action/iSingleRecord
- Action/iMultipleRecord
- Action/iAllRecord
After that we can make an abstract class : 
- AbstractAction
and a final : 
GenericAction
to force the extension of Abstract and not the Generic

i don't want to add only complexity, but i think that we need consistency and check over Type and Contracts is better than check over const.

Reply:
I thought about this, and probably can be done in the future, for now i'd like to keep options open.